Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStore.cpp

```diff
@@ -82,17 +82,26 @@ void CoordinatedBackingStore::paintToTextureMapper(TextureMapper& textureMapper,
                 && !tile.texture().filterOperation();
         }();
         if (canUseDamageToDrawTextureFragment) {
-            // We define damagedTileRect as a minimum bounding rectangle of all damage rects that intersect tile.rect()
-            // - this way we can keep a single texture draw call yet with potentially smaller sourceRect.
-            FloatRect damagedTileRect;
+            // We use Damage here as it's a convenient way to track per-tile damage.
+            // If the tile size is big enough, we use Damage::Mode::Rectangles so that
+            // more than one "draw" operation can be performed thus reducing the re-painted
+            // area significantly in many cases.
+            const bool shouldUseAtMostOneDrawCall = tile.rect().width() <= Damage::s_cellSize && tile.rect().height() <= Damage::s_cellSize;
+            IntRect tileIntRect = enclosingIntRect(tile.rect());
+            Damage tileDamage(tileIntRect, shouldUseAtMostOneDrawCall ? Damage::Mode::BoundingBox : Damage::Mode::Rectangles);
```

Comment:
I think we can optimize the case of using rectangles when the area is smaller than cell size to be equivalent to bounding box mode. We could check in unite() if the vector size is 1 and then just set the minimum bounding box at 0.

Reply:
I just noticed `tileIndexForRect` early returns 0 if `m_rects.size() == 1` so we're good